beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1247,48 +1268,130 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         }
     }
 
+    pub fn store_cold_state_summary(
+        &self,
+        state_root: &Hash256,
+        slot: Slot,
+        ops: &mut Vec<KeyValueStoreOp>,
+    ) -> Result<(), Error> {
+        ops.push(ColdStateSummary { slot }.as_kv_store_op(*state_root));
+        ops.push(KeyValueStoreOp::PutKeyValue(
+            get_key_for_col(
+                DBColumn::BeaconStateRoots.into(),
+                &slot.as_u64().to_be_bytes(),
+            ),
+            state_root.as_bytes().to_vec(),
+        ));
+        Ok(())
+    }
+
     /// Store a pre-finalization state in the freezer database.
-    ///
-    /// If the state doesn't lie on a restore point boundary then just its summary will be stored.
     pub fn store_cold_state(
         &self,
         state_root: &Hash256,
         state: &BeaconState<E>,
         ops: &mut Vec<KeyValueStoreOp>,
     ) -> Result<(), Error> {
-        ops.push(ColdStateSummary { slot: state.slot() }.as_kv_store_op(*state_root));
+        self.store_cold_state_summary(state_root, state.slot(), ops)?;
 
-        if state.slot() % self.config.slots_per_restore_point != 0 {
-            return Ok(());
+        let slot = state.slot();
+        match self.hierarchy.storage_strategy(slot)? {
+            StorageStrategy::ReplayFrom(from) => {
+                debug!(
+                    self.log,
+                    "Storing cold state";
```

Comment:
I quite like the uniformity of the logs because it helps with grepping things like "was there anything stored at this slot?". The `strategy: replay` makes it clear enough that a full state isn't being stored, at least for a debug log.